src/matlab2tikz.m

```diff
@@ -892,37 +892,48 @@ case guitypes()
 end
 % ==============================================================================
 function m2t = drawGridOfAxes(m2t, handle)
-    % draws the grids of an axes
-    %TODO: has{XYZ}Grid is always false without a good reason
-    hasXGrid = false;
-    hasYGrid = false;
-    hasZGrid = false;
-    if hasXGrid || hasYGrid || hasZGrid
-        matlabGridLineStyle = get(handle, 'GridLineStyle');
-        % Take over the grid line style in any case when in strict mode.
-        % If not, don't add anything in case of default line grid line style
-        % and effectively take Pgfplots' default.
-        defaultMatlabGridLineStyle = ':';
-        if m2t.cmdOpts.Results.strict ...
-                || ~strcmpi(matlabGridLineStyle,defaultMatlabGridLineStyle)
-            gls = translateLineStyle(matlabGridLineStyle);
-            axisGridOpts = {'grid style', sprintf('{%s}', gls)};
-            m2t.axesContainers{end}.options = cat(1, ...
-                m2t.axesContainers{end}.options,...
-                axisGridOpts);
-        end
-    else
+    % Draws the grids of an axis
+    gridOpts = opts_new();
+
+    % Check for major/minor grids
+    hasGrid =  [isOn(get(handle, 'XGrid'));
+                isOn(get(handle, 'YGrid'));
+                isOn(get(handle, 'ZGrid'))];
+
+    hasMinorGrid =  [isOn(get(handle, 'XMinorGrid'));
+                 	 isOn(get(handle, 'YMinorGrid'));
+                 	 isOn(get(handle, 'ZMinorGrid'))];
+
+    % Check for the grid options
+    if any(hasGrid)
+        % Get the line style and translate it to pgfplots
+        gridLS = translateLineStyle(get(handle, 'GridLineStyle'));
```

Comment:
I would prefer to keep having the check against the default value (and also the strict switch) here, e.g.
```matlab
[gridLS, isDefault] = getAndCheckDefault('axes', handle, 'GridLineStyle', ':');
if ~isDefault || m2t.Results.strict
    gridOpts = opts_add(gridOpts, ...);
end
```
 (note: code is untested).

That way, we keep our TikZ code leaner and we translate ugly MATLAB settings to beautiful TikZ ones (in HG2, I admit the situation has improved a lot, but in HG1 the dotted grid lines are a menace).